packages/snap-preact/src/Snap.tsx

```diff
@@ -237,6 +237,10 @@ export class Snap {
 			throw new Error(`Snap: config provided must contain a valid config.client.globals.siteId value`);
 		}
 
+		if (this.context.merchandising) {
+			this.config.client.globals.merchandising = this.context.merchandising;
```

Comment:
Do we want to pass everything off directly into the `globals.merchandising` here? It may be wise to just pick off the `segments` bit so that we are not locked into this `merchandising` context being only request params?